code/fireball/fireballs.cpp

```diff
@@ -407,10 +407,15 @@ void fireball_parse_tbl()
 	{
 		if (fi.lod_count > 1)
 		{
+			Assertion(fi.lod_count < MAX_FIREBALL_LOD, "Fireball LOD (%d) greater than MAX_FIREBALL_LOD %d.", fi.lod_count, MAX_FIREBALL_LOD);
+
 			auto lod0 = fi.lod[0].filename;
 
+			if (strlen(lod0) > MAX_FIREBALL_BASENAME_LEN)
+				Error(__FILE__, __LINE__, "A fireball base file name may not be longer than %d characters due to the LOD naming scheme. (is %s)", MAX_FIREBALL_BASENAME_LEN, lod0);
+
 			for (int j = 1; j < fi.lod_count; ++j)
-				sprintf(fi.lod[j].filename, "%s_%d", lod0, j);
+				sprintf(fi.lod[j].filename, "%.*s_%d", MAX_FIREBALL_BASENAME_LEN, lod0, j % MAX_FIREBALL_LOD);
```

Comment:
Instead of making a new #define which is only used here, change this to `MAX_FILENAME_LEN - 3`.  This makes explicit what is actually happening.

And remove `j % MAX_FIREBALL_LOD` since this obfuscates the assignment, and is unnecessary due to the newly added Assertion.

Reply:
`j % MAX_FIREBALL_LOD` is the only possible and single part of this commit necessary to make gcc12 understand that the preceding format string is indeed safe in this context. Adding comment to clarify.
The intention of the #define was to make the LOD naming scheme obvious outside of the implementation, but I can oblige with a magic number.

Reply:
That's odd, but ok.

Magic numbers should be assessed on a case-by-case basis.  In this case, the need for the number is dependent on the format string, and the entire context is all on one line.  So I don't think a separate #define, let alone spreading out the context over multiple files, is the right thing to do.

Reply:
Or, hmm, the context is in one scoped block, rather than one line.  If you wanted to put a `constexpr` above line 414, that would work.